### PR TITLE
libsass: add mingw support and pure C consumer + del fPIC if shared + early checks in configure() + explicit pkg_config name

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **libsass/all**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libsass/all/conanfile.py [hunk 1]
```diff
@@ -14,18 +16,24 @@ class LibsassConan(ConanFile):
...
 
     @property
     def _source_subfolder(self):
         return "source_subfolder"
 
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
     def validate(self):
```
Comment (ericLemanissier) [inline recipes/libsass/all/conanfile.py:29]:
```suggestion
```
This should work around this bug: https://github.com/conan-io/conan-center-index/pull/4182#issuecomment-758889526

Comment (SpaceIm) [reply recipes/libsass/all/conanfile.py:29]: yes, we see the advantage to keep early `ConanInvalidConfiguration` in `configure()`.

Is `sassc` compatible with Windows? Shouldn't you raise in `sassc` recipe also?

Comment (ericLemanissier) [reply recipes/libsass/all/conanfile.py:29]: I'll raise the exception in sassc too.
oth recipe can be made compatible with windows, it should be buildable with the provided visual studio solution, but I have no need for it and no time to work on this, so maybe later

File: recipes/libsass/all/conanfile.py [hunk 2]
```diff
@@ -2,6 +2,8 @@
 from conans.errors import ConanInvalidConfiguration
 import os
 
+required_conan_version = ">=1.32.0"
+
```
Comment (ericLemanissier) [inline recipes/libsass/all/conanfile.py:6]:
```suggestion
```
this is not necessary if the other change is applied

Comment (SpaceIm) [inline recipes/libsass/all/conanfile.py:6]:
```suggestion
```

Comment (SpaceIm) [inline recipes/libsass/all/conanfile.py:6]:
```suggestion
```

**Review by ericLemanissier — DISMISSED** [review]

File: recipes/libsass/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/libsass/all/conanfile.py]: I'll try to support MinGW in libsass, so only Visual Studio will be unsupported for the moment.

**Review by mathbunnyru — APPROVED** [review]

**Review by mathbunnyru — APPROVED** [review]

**Review by ericLemanissier — APPROVED** [review]

File: recipes/libsass/all/conanfile.py [hunk 3]
```diff
@@ -43,21 +58,64 @@ def _configure_autotools(self):
...
+    def _install_autotools(self):
         with tools.chdir(self._source_subfolder):
             autotools = self._configure_autotools()
             autotools.install()
-        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
         tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
         tools.remove_files_by_mask(self.package_folder, "*.la")
 
+    def _install_mingw(self):
+        self.copy("*.h", dst="include", src=os.path.join(self._source_subfolder, "include"))
```
Comment (SSE4) [inline recipes/libsass/all/conanfile.py:104]: so, `make install` doesn't work for MinGW?

Comment (SpaceIm) [reply recipes/libsass/all/conanfile.py:104]: Generally speaking, it should work, but install target calls something that didn't work on my computer. Moreover it installs runtime in lib folder, so it's just easier to manually copy these files.

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**